### ISSUE TITLE
Duplicate file entries

### ISSUE THREAD

**Issue by jeremypw** [elementary/files#2190]
### What Happened?

Since https://github.com/elementary/files/commit/db6dfad35059bcd9a0a3ea1401a42c59f3d8942a some entries in the view have appeared duplicated.

### Steps to Reproduce

1. Show a folder in e.g. ListView containing a significant number of folders and files (>20 ?)
2. Observe some files have more than one entry.

### Expected Behavior

One entry per file.

### OS Version

7.x (Horus)

### Software Version

Compiled from git

**Comment by peteruithoven** [comment]
I couldn't really reproduce this one like that. But I did notice that when I created files outside of Files, for example using `touch ...` in the Terminal those files appeared twice.

**Comment by jeremypw** [comment]
Yes, I noticed that some files were duplicated and some were not and I haven't determined exactly why.  Maybe a timing issue.
